s3/pkg/datastore/azure/azure.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	log "github.com/sirupsen/logrus"
+	backendpb "github.com/opensds/multi-cloud/backend/proto"
+	. "github.com/opensds/multi-cloud/s3/error"
+	dscommon "github.com/opensds/multi-cloud/s3/pkg/datastore/common"
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+)
+
+// TryTimeout indicates the maximum time allowed for any single try of an HTTP request.
+var MaxTimeForSingleHttpRequest = 50 * time.Minute
+
+type AzureAdapter struct {
```

Comment:
please share the test report and test cases tested for all backends, like AWS, AZURE, GCP etc. Thanks,.

Reply:
thanks @rhsakarpos for your suggestion. But currently, the data storage layer is not finished yet. According to this PR, this only contains the bucket creation and object put logic, so, we cannot pass all the test cases for aws, azure, gcp. But for object put in yig data storage layer, the related test cases are passed. Do you need this one?

Reply:
@nguptaopensds please confirm, i was under the understanding that the features that come in the YIG PR should be working with ALL cloud backends, like Azure, AWS, GCP and HWS...do correct me if i am mistaken. thanks